File: mlt++/swig/ruby/play.rb
#!/usr/bin/env ruby

# Import required modules
require 'mltpp'

# Create the mlt system
Mltpp::Factory::init

# Establish the mlt profile
profile = Mltpp::Profile.new

# Get and check the argument
file = ARGV.shift
raise "Usage: test.rb file" if file.nil?

# Create the producer
producer = Mltpp::Factory::producer( profile, file )
raise "Unable to load #{file}" if !producer.is_valid

# Create the consumer
consumer = Mltpp::Consumer.new( profile, "sdl" )
raise "Unable to open sdl consumer" if !consumer.is_valid

# Turn off the default rescaling
consumer.set( "rescale", "none" )

# Set up a 'wait for' event
event = consumer.setup_wait_for( "consumer-stopped" )

# Start the consumer
consumer.start

# Connect the producer to the consumer
consumer.connect( producer )

# Wait until the user stops the consumer
consumer.wait_for( event )

# Clean up consumer
consumer.stop

File: mlt++/swig/python/play.py
#!/usr/bin/env python

# Import required modules
import mltpp
import time
import sys

# Start the mlt system
mltpp.Factory.init( )

# Create the producer
p = mltpp.Producer( sys.argv[1] )

if p:
	# Create the consumer
	c = mltpp.Consumer( "sdl" )

	# Turn off the default rescaling
	c.set( "rescale", "none" )
	
	# Connect the producer to the consumer
	c.connect( p )
	
	# Start the consumer
	c.start( )
	
	# Wait until the user stops the consumer
	while c.is_stopped( ) == 0:
		time.sleep( 1 )
else:
	# Diagnostics
	print "Unable to open ", sys.argv[ 1 ]

